service/src/main/java/com/ge/predix/acs/privilege/management/dao/SubjectMigrationManager.java

```diff
@@ -0,0 +1,41 @@
+package com.ge.predix.acs.privilege.management.dao;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+
+@Component
+@Profile("titan")
+public class SubjectMigrationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubjectMigrationManager.class);
+
+    public void doSubjectMigration(final SubjectRepository subjectRepository,
+            final GraphSubjectRepository subjectHierarchicalRepository, final int pageSize) {
+        int numOfSubjectsSaved = 0;
+        Pageable pageRequest = new PageRequest(0, pageSize, new Sort("id"));
+        long numOfSubjectEntitiesToMigrate = subjectRepository.count();
+        Page<SubjectEntity> pageOfSubjects;
+
+        do {
+            pageOfSubjects = subjectRepository.findAll(pageRequest);
+            List<SubjectEntity> subjectListToSave = pageOfSubjects.getContent();
+            numOfSubjectsSaved += pageOfSubjects.getNumberOfElements();
+            subjectListToSave.forEach(item -> item.setId(0));
```

Comment:
@sanjeevchopra ,

We should do the logging here for the migrated subject similar to how it's done for the resources above in ResourceMigrationManager.java

`
subjectListToSave.stream().peek(item -> LOGGER.trace("doSubjectMigration Subject-Id : " + item.getSubjectIdentifier() + " Zone-name : " + item.getZone().getName() + " Zone-id:" + item.getZone().getId())).forEach(item -> item.setId((long) 0));
`